test/server/echo.go

```diff
@@ -120,3 +143,61 @@ func main() {
 	signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM)
 	<-sigs
 }
+
+func setResponseFromCodes(request *http.Request, response http.ResponseWriter) error {
+	responseCodes := request.FormValue("codes");
+
+	codes, err := validateCodes(responseCodes)
+	if err != nil {
+		return err
+	}
+
+	// Keep a cursor for each sequence of response codes
+	position, ok := state[responseCodes]
+	if !ok {
+		state[responseCodes] = 0
+	}
+	if len(codes) > 0 {
+		state[responseCodes] = (position + 1) % len(codes)
+	}
+
+	response.WriteHeader(codes[position])
+	return nil
+}
+
+
+// codes must be comma-separated HTTP response codes
+func validateCodes(codestrings string) ([]int, error) {
+
+	if codestrings == "" {
+		// Consider no codes to be a repeating sequence of 200,
+		codestrings = strconv.Itoa(http.StatusOK)
+	}
+
+	aCodestrings := strings.Split(codestrings, ",")
+	codes := make([]int, len(aCodestrings), len(aCodestrings))
+
+	for i, codestring := range aCodestrings {
+		code, err := validateCode(codestring)
```

Comment:
I think it would be simpler to maintain a global request count and then just do a modulus in the request array.